### ISSUE TITLE
Graph Tab Empty  Cacti 1.2.10

### ISSUE THREAD

**Issue by thereporter42** [Cacti/cacti#3400]
**Describe the bug**
This was happening is 1.2.8 and the ticket was closed
but I am running 1.2.10 and I am getting the error 
in this version. 

Here is the old ticket. 
https://github.com/Cacti/cacti/issues/3140

I am getting the same issue the cacti server is running on Debian 10.3 here 
and it is Cacti 1.2.10

root@server1:/etc# cat /etc/debian_version
10.3

 uname -a
Linux server1 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux

php -v
PHP 7.3.14-1~deb10u1 (cli) (built: Feb 16 2020 15:07:23) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.14, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.14-1~deb10u1, Copyright (c) 1999-2018, by Zend Technologies

apache2 -v
Server version: Apache/2.4.38 (Debian)
Server built:   2019-10-15T19:53:42

![image](https://user-images.githubusercontent.com/38896397/78419906-9aab7d00-7607-11ea-9ff7-50e7f7d8b55c.png)

Now in this picture I am using Firefox 74.0.1 (64 bit)  on Windows 10 (64 bit) to access the cacti
server. 

Ghostery was disabled as well during this test! (ad blocker) so it isn't that 
and it is doing the same thing in Chrome as well which has no blocker.

Tried it in Chrome 80.0.3987.163 (Official Build) (64 Bit)  same results the 
Graph tab is empty there too so I doubt it is a browser issue trying two different
browsers both with the same result.  I tried both Chrome and Firefox

Here it is in Chrome with the same problem 

![image](https://user-images.githubusercontent.com/38896397/78420270-c5e39b80-760a-11ea-815c-5e7d0eb91aae.png)

Again the graphs are there if I go under console, management, graphs I can see the graphs there
with data on them but they just don't show up on the graph tab at the top when
I click on it. 

![image](https://user-images.githubusercontent.com/38896397/78420030-c418d880-7608-11ea-9094-0f66ca014484.png)

So why is the graph tab up at the top not working? 

I believe this is the same error. 

What is the fix for it? I am running Cacti 1.2.10 and I have tried
both the Modern Theme and the the Dark Theme it 
doesn't matter I am getting the same result with no graphs
showing up on my graph tab. 

So what is the fix for this? 

**To Reproduce**
Unknown

**Expected behavior**

That the graphs would show up in the tab 

**Screenshots**
See above in the description I have included screenshots

**Desktop (please complete the following information):**
 -  Windows 10 Pro (64 bit) 
 - Browser is Firefox 
 - Version 74.0.1 (64 bit) 

**Smartphone (please complete the following information):**
N/A not using smart phone

**Additional context**

**Comment by netniV** [comment]
If you open up the developer console, do you have any errors?  Also, I noticed you were on Debian, so I wonder if this is an external package related issue but it shouldn't be or we'd see more issues.

**Comment by netniV** [comment]
Actually, having seen your other post, I think you may be experiencing that specific issue and need the latest version (1.2.11) that will be released this weekend.

**Comment by thereporter42** [comment]
What specific issue that other ticket is closed for version 1.2.8 which means it would have been
resolved. 

This isn't a duplicate ticket at all this is a new issue. 


I am looking at my log file and I am not seeing any errors per say I have one warning but it 
wouldn't affect the graph tab it happened well after I made this ticket. 

2020/04/04 01:00:36 - CMDPHP WARNING: Data input script not found in input whitelist file: /usr/local/etc/cacti/input_whitelist.json 

The logs can be seen at 

https://pastebin.com/0ATE34U4

**Comment by TheWitness** [comment]
Can you please redo your test with the 1.2.x branch?  Thanks.

**Comment by thereporter42** [comment]
What do you mean? That is the latest! You have not released 1.2.11 yet.

**Comment by TheWitness** [comment]
I suspect this is fixed there.  The work is pretty much done, we just have not released it.  We would like to get a few people to test it.

To download the 1.2.x branch. just go-to the 'Code' area, and from the "branch" pulldown, select 1.2.x.  Of if you are using the git cli, you can do the following:

```
git clone -b 1.2.x https://github.com/Cacti/cacti/cacti.git tosomelocaiton
```

**Comment by thereporter42** [comment]
root@server1:/opt# git clone -b 1.2.x https://github.com/Cacti/cacti/cacti.git /opt/cacti
Cloning into '/opt/cacti'...
remote: Not Found
fatal: repository 'https://github.com/Cacti/cacti/cacti.git/' not found

Doesn't work

**Comment by TheWitness** [comment]
Did I do a typo.  Would not be the first time ;)

**Comment by TheWitness** [comment]
https://github.com/Cacti/cacti.git

**Comment by thereporter42** [comment]
That is what I did before and I get 1.2.10 and that is the issue I am having issues with!

**Comment by TheWitness** [comment]
Don't forget the -b 1.2.x though.
That pulls a completely different version down.

**Comment by thereporter42** [comment]
The line you gave me does NOT work! If you put the -b 1.2.x it doesn't work


git clone -b 1.2.x https://github.com/Cacti/cacti/cacti.git /opt/cacti
Cloning into '/opt/cacti'...
remote: Not Found
fatal: repository 'https://github.com/Cacti/cacti/cacti.git/' not found

**Comment by TheWitness** [comment]
Oh, geeze, man dude, I've got you all mixed up.  This is how "I" upgrade Cacti

```cmd
rm -rf cacti-develop
git clone -b 1.2.x https://github.com/Cacti/cacti.git cacti-develop
/bin/cp -rpf cacti-develop/* cacti
chown -R apache:apache cacti
```

**Comment by TheWitness** [comment]
I hope that makes better sense...

**Comment by thereporter42** [comment]
That line worked 
git clone -b 1.2.x https://github.com/Cacti/cacti.git cacti-develop

**Comment by TheWitness** [comment]
Okay, keep us posted if it indeed fixed the issue.

**Comment by thereporter42** [comment]
Ok so I am running the version you posted which is Cacti 1.2.11 and I still have
the same issue as before no graphs are showing up in the Graph tab but it did fix the
Search box though. 

Firefox 74.0.1 (64 bit) 

![image](https://user-images.githubusercontent.com/38896397/78454436-331c1e80-7655-11ea-9759-7bcd27488b0f.png)

Also tried Chrome 80.0.3987.163 (Official Build) (64 Bit) 

Same problem here as well !

**Comment by TheWitness** [comment]
So, use F12 to open the debug window.  Then, goto the Console tab of the Debug Window, refresh the page and take a screen grab of the errors there.

**Comment by TheWitness** [comment]
Also, is this a Windows machine?  Can you see if you have any Tree's defined.  Goto Console > Management > Trees.

**Comment by TheWitness** [comment]
If you don't have any Tree's, create one, put a host on it and see if the problem goes away.

**Comment by TheWitness** [comment]
Sorry, just noted "debian".

**Comment by thereporter42** [comment]
I am using Debian 10.3 not windows that is evil I wouldn't run a server with Windows
that just sucks. 


So I hit F12 and then I hit F5 to refresh the page and this is what
I got 

Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. graph_view.php
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.


Looks like child-src is depreciated also seems to be an issue with zoom as well

**Comment by TheWitness** [comment]
Well, that's odd, cause we don't define 'child-src', and we don't use 'frame-src'.  Those are warnings anyway.  Did you get any errors?

**Comment by TheWitness** [comment]
Run that upgrade script again, and refresh the Graphs page using an F5.

**Comment by thereporter42** [comment]
And when I check the CSS and XHR at the bottom it spits me out more errors. 

well you must be defining it some where because it is being used in Cacti. 
They are warnings but may be affecting it.  Run what upgrade script 
I have the version installed you want installed. 

Here are some errors again this is pointing to a CSS error some where 
in Cacti. 


Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. graph_view.php
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
Error in parsing value for ‘background’.  Declaration dropped. graph_view.php:1:96
Expected ‘none’, URL, or filter function but found ‘progid’.  Error in parsing value for ‘filter’.  Declaration dropped. graph_view.php:1:150
Error in parsing value for ‘align-self’.  Declaration dropped. main.css:322:14
Unknown property ‘-moz-border-radius’.  Declaration dropped. main.css:905:21
Unknown property ‘-moz-border-radius’.  Declaration dropped. main.css:937:21
Expected color but found ‘show’.  Error in parsing value for ‘color’.  Declaration dropped. main.css:1127:9
Expected color but found ‘-webkit-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1292:33
Expected color but found ‘-o-linear-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1293:35
Expected color but found ‘-moz-linear-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1294:37
Expected color but found ‘-webkit-linear-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1295:40
Expected color but found ‘-ms-linear-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1296:36
Expected color but found ‘linear-gradient’.  Error in parsing value for ‘border-color’.  Declaration dropped. main.css:1297:32
Unknown property ‘-moz-border-radius-topright’.  Declaration dropped. main.css:1883:30
Unknown property ‘-moz-border-radius-topleft’.  Declaration dropped. main.css:1884:29
Unknown property ‘-moz-border-radius’.  Declaration dropped. main.css:1955:21
Unknown property ‘zoom’.  Declaration dropped. main.css:1995:7
Error in parsing value for ‘float’.  Declaration dropped. main.css:2268:9
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2406:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2407:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2408:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2409:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2495:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2496:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2497:14
Error in parsing value for ‘box-shadow’.  Declaration dropped. main.css:2498:14
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery.zoom.css:10:9
Unknown property ‘-moz-opacity’.  Declaration dropped. jquery.zoom.css:11:15
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery.zoom.css:34:9
Unknown property ‘-moz-opacity’.  Declaration dropped. jquery.zoom.css:35:15
Unknown property ‘-moz-box-shadow’.  Declaration dropped. jquery.zoom.css:44:18
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery.zoom.css:54:9
Unknown property ‘-moz-opacity’.  Declaration dropped. jquery.zoom.css:55:15
Unknown property ‘-moz-box-shadow’.  Declaration dropped. jquery.zoom.css:65:18
Expected ‘none’, URL, or filter function but found ‘Alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:52:9
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:680:10
Expected ‘none’, URL, or filter function but found ‘Alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:1044:9
Expected ‘none’, URL, or filter function but found ‘Alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:1051:9
Expected ‘none’, URL, or filter function but found ‘Alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:1055:9
Expected ‘none’, URL, or filter function but found ‘Alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery-ui.css:1307:10
Unknown property ‘-moz-box-shadow’.  Declaration dropped. style.css:116:19
Unknown property ‘-moz-text-shadow’.  Declaration dropped. style.css:144:20
Unknown property ‘-moz-border-radius’.  Declaration dropped. style.css:147:22
Unknown property ‘-moz-box-shadow’.  Declaration dropped. style.css:154:19
Unknown property ‘-moz-box-shadow’.  Declaration dropped. style.css:169:19
Unknown property ‘-moz-text-shadow’.  Declaration dropped. style.css:184:20
Elements matching selector: .vakata-context .vakata-context-separator > a, .vakata-context .vakata-context-separator > a:hover
NodeList []
Unknown property ‘-moz-box-shadow’.  Declaration dropped. style.css:187:19
Unknown property ‘-moz-border-radius’.  Declaration dropped. style.css:190:22
Expected ‘none’, URL, or filter function but found ‘alpha(’.  Error in parsing value for ‘filter’.  Declaration dropped. jquery.colorpicker.css:233:14
Unknown property ‘-moz-box-shadow’.  Declaration dropped. c3.css:112:19
Elements matching selector: .c3-tooltip
NodeList []
Unknown property ‘-moz-osx-font-smoothing’.  Declaration dropped. fontawesome.css:10:27

**Comment by TheWitness** [comment]
Yea, that CSS stuff is garbage.  It's that beautiful world of Browser compatibility.  Hopefully, we can drop all the 'custom' attributes in the next year or so.  We should drop the "filter" stuff since that is like a pre ie10 gradient syntax.  None of those errors should stop the tree from rendering though.

What OS holds the browser?  Can you test from an OS with an earlier browser?

**Comment by thereporter42** [comment]
This is Windows 10 Running FireFox 74.0.1 (64 bit) 

Well basically I don't know then why I am getting no graphs showing up
how do I download 1.2.8 I am tired of this and I need a working 
cacti version. 

I still think those CSS errors are causing it to not display the graphs.

**Comment by TheWitness** [comment]
Okay, I'm running 68.6.1esr (64bit) on Windows 10.  Let me upgrade and see if the problem happens for me there as well.

**Comment by TheWitness** [comment]
Oh, this new Firefox is a dog, it's offloading all DNS requests to a Cloudflare DNS (cause it loves you).  Man, that's making my pages load for shit.  Going to try to disable that right now.

**Comment by TheWitness** [comment]
Yea, it's a total dog.  It's for your protections you understand right?  That's why they ran the former CEO out on a rail cause he was able to distinguish between a man a woman, and a well, let's say 'other'.

**Comment by TheWitness** [comment]
Well, I disabled the DNS and it's still a dog on the Tree page.  In fact, it's a dog.  Ca you go to the 68.6.1esr version and see if it starts working?

**Comment by thereporter42** [comment]
No I can't

**Comment by TheWitness** [comment]
Well, I am.  It's a total dog.  You should report this to Mozilla okay?

**Comment by TheWitness** [comment]
https://ftp.mozilla.org/pub/firefox/releases/68.6.1esr/win64/en-US/

**Comment by TheWitness** [comment]
I did find a few things like liked in the 74 release like styling of the scroll bar.  But performance was horrible.

**Comment by TheWitness** [comment]
In case you loose you mind, and want to go back, here is what you need to read/follow.

https://support.mozilla.org/en-US/kb/recovering-important-data-from-an-old-profile

**Comment by TheWitness** [comment]
I'm going to consider this a Firefox issue.  Please use a different browser and let me know if the page is working.  If so, I'll go ahead and close this one.

**Comment by thereporter42** [comment]
I am not reporting it to Firefox this is something in your release it doesn't work in CHROME either
so with it being too browsers this is something in Cacti!  

This is a CACTI Issue! Please resolve it!

**Comment by TheWitness** [comment]
I see no issues with Cacti ATM, and I just re-installed.  Sorry, you will have to troubleshoot and let us know what you find.

**Comment by thereporter42** [comment]
clearly there is an issue with cacti because this won't work in either Chrome or Firefox 
simply because you don't seem to like Firefox you are telling me to report it to them instead
of you dealing with this issue. 

The errors I posted above in the CSS shows that there IS issues with Cacti 
in the CSS file.

**Comment by thereporter42** [comment]
Funny I went back to 1.2.9 and it works fine I just tested it.  Using the same
version of Firefox I was running before. 

![image](https://user-images.githubusercontent.com/38896397/78456501-4cc36300-7661-11ea-9860-9cea58e231d0.png)


https://github.com/Cacti/cacti/releases/tag/release%2F1.2.9

This ticket should not be closed you have a valid issue with 1.2.10 and 1.2.11 

I want to speak to the head dev for this project.

**Comment by TheWitness** [comment]
When you are paying for support, you can ask those questions, otherwise, you are just going to piss off me, and everyone else who does this in their spare time.  Also, we don't offer paid support.  I'll give you one more shot before I just ignore you.  Update again to 1.2.11, and then try and access the site from https instead of http.

**Comment by TheWitness** [comment]
Also, check your apache error log's for clues.  You can also enable a static error log by editing the php.ini (there will be two of them on debian) and uncommenting error_log and redirecting to /tmp/error_log.log.  Note that systemd will place that /tmp dir under /tmp/systmed*httpd/tmp or something similar.

**Comment by TheWitness** [comment]
It might even be something like SELinux.  But on CentOS7, PHP54, there are presently no issues, even with FF74 being a dog, it eventually renders the page, if not simply just real slow.

**Comment by thereporter42** [comment]
There are no errors in the apache log the  issue is NOT on my side.  
There are no errors in my apache.log 

This is an error in your main.css style sheet the errors 
are quite clear. 

I am helping you trouble shoot this for NOTHING I am not getting paid
to troubleshoot this issue.  I am here troubleshooting your issues 
with your project for nothing. So don't talk to me about pissing people
off. 

I reported a problem to you and you need to fix it I won't be going
back to 1.2.10 or 1.2.11 as I have a working version right now. 

There is no issues with FireFox 74 at all it works fine you simply don't like
it TheWitness so you just closed off the ticket even though there are errors
in your CSS file. 

The graphs never show up on my end so I dunno what you are 
talking about rendering. 

I believe the errors in the CSS files in 1.2.10 and 1.2.11 are stopping
the graphs from being displayed. You have added something and it broke
in 1.2.10

**Comment by TheWitness** [comment]
Okay, then do this.  Take the CSS files from 1.2.9.  Since you are using the Dark them, that's just one file main.css, copy it over to the 1.2.11 install, and if it starts working, I'll eat my lunch, and it's just about lunch time.  If it still doesn't work, you need to donate some money, or just apologize for being an arse.

**Comment by cigamit** [comment]
Locking conversation, this is a free software, there is no reason to be at rude to people who are doing a ton of work for no pay.

**Comment by netniV** [comment]
I would have to agree with @cigamit on this.  Lately, there seems to be an increase in people who get into unnecessarily heated states whilst we are trying to help.

Just to add, I have also used Firefox and Chrome across multiple installations and not experienced the issue of no graphs appearing.  I suspect this is linked to the CSS styling issue that is already resolved and will be released more than likely tomorrow in 1.2.11.

And one final thing is, just because you have a problem, doesn't mean we can just fix it.  We have to know what the issue is and understand why it is broken for you.  No one disputes that someone has a problem, we just may not always be able to replicate and thus understand what is needed to fix it.
